torch/testing/_internal/common_modules.py

```diff
@@ -1399,6 +1399,15 @@ def module_inputs_torch_nn_LSTM(module_info, device, dtype, requires_grad, train
                skips=(
                    DecorateInfo(skipIfMps, 'TestModule', dtypes=[torch.float64]),),
                supports_gradgrad=False),
+    # TransformerEncoder takes the same inputs as TransformerEncoderLayer
+    ModuleInfo(torch.nn.TransformerEncoder,
```

Comment:
Here, this should work for you:

```python
def module_inputs_torch_nn_TransformerEncoder(module_info, device, dtype, requires_grad, training, **kwargs):
    # Reuse the TransformerEncoderLayer samples since the forward args are nearly the same.
    for layer_module_input in module_inputs_torch_nn_TransformerEncoderLayer(
            None, device, dtype, requires_grad, training):
        # Construct a TransformerEncoderLayer object to pass to TransformerEncoder.
        l_args, l_kwargs = (layer_module_input.constructor_input.args,
                            layer_module_input.constructor_input.kwargs)
        encoder_layer = torch.nn.TransformerEncoderLayer(*l_args, **l_kwargs)
        num_layers = 2
        # Note: TransformerEncoderLayer takes a "src_mask" while
        # TransformerEncoder takes a "mask"; rename kwarg appropriately.
        forward_input = layer_module_input.forward_input
        if 'src_mask' in forward_input.kwargs:
            forward_input.kwargs['mask'] = forward_input.kwargs['src_mask']
            del forward_input.kwargs['src_mask']
        yield ModuleInput(
            constructor_input=FunctionInput(encoder_layer, num_layers),
            forward_input=forward_input,
            desc=layer_module_input.desc
        )

...

# ... in the module_db list ...
ModuleInfo(torch.nn.TransformerEncoder,
           train_and_eval_differ=True,
           module_inputs_func=module_inputs_torch_nn_TransformerEncoder,
           skips=(
               # No channels_last support for TransformerEncoderLayer currently.
               DecorateInfo(unittest.skip("Skipped!"), 'TestModule', 'test_memory_format'),
               # Doesn't support device / dtype kwargs directly because it is just a
               # container of TransformerEncoderLayers.
               DecorateInfo(unittest.expectedFailure, 'TestModule', 'test_factory_kwargs'),
               DecorateInfo(skipIfMps, 'TestModule', dtypes=[torch.float64]),)
           ),
```

Reply:
Thanks, this worked.